docs/Debugging.md

```diff
@@ -0,0 +1,58 @@
+# Debugging Goblint
+
+## Building a Debuggable Goblint Executable
+
+To build a Goblint executable with debug information, run the following command within the `analyzer` directory.
+
+`make debug`
```

Comment:
I haven't gotten around to trying earlybird but when I read about it, it required OCaml bytecode compilation. I see this debug target does that:
https://github.com/goblint/analyzer/blob/11dcafdfd30fac8ba089840a633461ee08ddf3d1/make.sh#L61-L63

Unlike the normal native build, this one still uses ocamlbuild instead of dune. Although this isn't necessarily about the documentation, I'm wondering if we could easily change this target to also build using dune.